### PR TITLE
Connect to services over dmsghttp

### LINKED ISSUE #985: Add dmsghttp support
**Feature description**

We should add `dmsghttp` support to the visor, allowing to make `http` requests via dmsg if the user choses. All services will listen on dmsg and plain tcp instead of a plain tcp only.

### PR DESCRIPTION
**Pull request by ersonp**
Did you run `make format && make check`?
yes

Fixes #985 

 Changes:	
- Updated `disc.NewHTTP`
- Added `http.Client` param to `dmsgc.New()`
- Update `initDmsg()` to connect to the `dmsg-discovery` via HTTP over dmsg if the URL of `dmsg-discovery` in the config is `dmsg://<pk-of-dmsg-discovery>:80`
- Added `dmshhttp` module
- Renamed `localServers.json` to `dmsghttp-config.json`
- Changed datatype of `DMSGServers` from `[]string` to `[]*disc.Entry`

How to test this PR:

On test deployment:
1. Run
    ```
    ./skywire-cli config gen -dirt
    ```
2. Run 
    ```
    ./skywire-visor skywire-config.json
    ```
3. Open hypervisor UI and check functionality. 

On integration env:
Check [#skywire-services-486](https://github.com/SkycoinPro/skywire-services/pull/486)

Needs:
[#dmsg-130](https://github.com/skycoin/dmsg/pull/130)

### CODE REVIEW

**Review by mrpalide — APPROVED** [review]
Code is clean and actually work both test, in integration env and locally independent.

File: pkg/dmsgc/dmsgc.go [hunk 1]
```diff
@@ -13,13 +14,13 @@ import (
...
-	SessionsCount int      `json:"sessions_count"`
-	Servers       []string `json:"servers"`
+	Discovery     string        `json:"discovery"`
+	SessionsCount int           `json:"sessions_count"`
+	Servers       []*disc.Entry `json:"servers"`
 }
 
 // New makes new dmsg client from configuration
-func New(pk cipher.PubKey, sk cipher.SecKey, eb *appevent.Broadcaster, conf *DmsgConfig) *dmsg.Client {
+func New(pk cipher.PubKey, sk cipher.SecKey, eb *appevent.Broadcaster, conf *DmsgConfig, httpC http.Client) (dmsgC *dmsg.Client) {
```
Comment (jdknives) [inline pkg/dmsgc/dmsgc.go]: Why the named return here?

Comment (ersonp) [reply pkg/dmsgc/dmsgc.go]: Changed it.

File: cmd/skywire-cli/commands/config/gen.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ var genConfigCmd = &cobra.Command{
 		// Use local servers
 		if dmsgHTTP {
 			var dmsgHTTPServersList visorconfig.DmsgHTTPServers
-			serversListJSON, err := ioutil.ReadFile("localServers.json")
+			serversListJSON, err := ioutil.ReadFile("dmsghttp-config.json")
 			if err != nil {
 				logger.WithError(err).Fatal("Failed to read servers.json file.")
```
Comment (jdknives) [inline cmd/skywire-cli/commands/config/gen.go:112]: Comment needs to be updated.

**Review by jdknives — APPROVED** [review]
Looks good. Just one small point.

**Pull request merged**